src/axom/primal/tests/primal_clip.cpp

```diff
@@ -434,46 +439,63 @@ void check_hex_tet_clip(double EPS)
   HexahedronType* hex = axom::allocate<HexahedronType>(1);
   PolyhedronType* res = axom::allocate<PolyhedronType>(1);
 
-  tet[0] = TetrahedronType(PointType {1, 0, 0},
-                           PointType {1, 1, 0},
-                           PointType {0, 1, 0},
-                           PointType {1, 0, 1});
-
-  hex[0] = HexahedronType(PointType {0, 0, 0},
-                          PointType {1, 0, 0},
-                          PointType {1, 1, 0},
-                          PointType {0, 1, 0},
-                          PointType {0, 0, 1},
-                          PointType {1, 0, 1},
-                          PointType {1, 1, 1},
-                          PointType {0, 1, 1});
+  // Shapes on host
+  TetrahedronType tet_host;
+  HexahedronType hex_host;
+  PolyhedronType res_host;
+
   axom::for_all<ExecPolicy>(
     1,
-    AXOM_LAMBDA(int i) { res[i] = axom::primal::clip(hex[i], tet[i]); });
+    AXOM_LAMBDA(int i) {
+      tet[0] = TetrahedronType(PointType {1, 0, 0},
+                               PointType {1, 1, 0},
+                               PointType {0, 1, 0},
+                               PointType {1, 0, 1});
+
+      hex[0] = HexahedronType(PointType {0, 0, 0},
+                              PointType {1, 0, 0},
+                              PointType {1, 1, 0},
+                              PointType {0, 1, 0},
+                              PointType {0, 0, 1},
+                              PointType {1, 0, 1},
+                              PointType {1, 1, 1},
+                              PointType {0, 1, 1});
```

Comment:
Does this change require `setDefaultAllocatorID()`  to have been set above? Or does it work because its constructed within the appropriate `ExecPolicy` ?

If it's the former, is there a way to make this work without setting the default allocator?

Reply:
> Does this change require setDefaultAllocatorID() to have been set above? Or does it work because its constructed within the appropriate ExecPolicy ?

The latter, and it works because the hexahedron constructor is host-device decorated.